File: scripts/export-geographic-dictionary-text.mjs
import { readFile, writeFile } from "node:fs/promises";
import path from "node:path";
import pdf from "pdf-parse";

const PDF_PATH = path.resolve(
  "data/Geographic Dictionary of the Virgin Islands.pdf"
);
const OUTPUT_PATH = path.resolve("data/geographic-dictionary.txt");

async function main() {
  const buffer = await readFile(PDF_PATH);
  const result = await pdf(buffer);

  if (!result.text || !result.text.trim()) {
    throw new Error("No text extracted from PDF.");
  }

  await writeFile(OUTPUT_PATH, result.text, "utf8");
  console.log(`Wrote text export to ${OUTPUT_PATH}`);
}

main().catch((error) => {
  console.error("PDF export failed:", error);
  process.exit(1);
});

File: scripts/export_geographic_dictionary_text.py
from pathlib import Path
from pypdf import PdfReader

ROOT = Path.cwd()
pdf_path = ROOT / "data" / "Geographic Dictionary of the Virgin Islands.pdf"
out_path = ROOT / "data" / "geographic-dictionary.txt"

reader = PdfReader(str(pdf_path))

chunks = []
for i, page in enumerate(reader.pages, start=1):
    text = page.extract_text() or ""
    chunks.append(f"<PARSED TEXT FOR PAGE: {i} / {len(reader.pages)}>\n{text}\n")

out_path.parent.mkdir(parents=True, exist_ok=True)
out_path.write_text("\n".join(chunks), encoding="utf-8")

print(f"Wrote {out_path}")